index.js

```diff
@@ -1,21 +1,57 @@
-const marked = require('marked');
+const checkOutstandingTasks = require('./src/check-outstanding-tasks');
 
 module.exports = (app) => {
   app.log('Yay! The app was loaded!');
 
   // watch for pull requests & their changes
-  app.on(['pull_request.opened', 'pull_request.edited', 'pull_request.synchronize'], async context => {
+  app.on([
+    'pull_request.opened',
+    'pull_request.edited',
+    'pull_request.synchronize',
+    'issue_comment', // for comments on github issues
+    'pull_request_review_comment', // comment lines on diffs for reviews
+  ], async context => {
     const startTime = (new Date).toISOString();
 
-    // lookup the pr body/description
-    const pr = context.payload.pull_request;
-    const body = pr.body;
+    // lookup the pr
+    let pr = context.payload.pull_request;
 
-    const tokens = marked.lexer(body, { gfm: true });
-    const listItems = tokens.filter(token => token.type === 'list_item_start');
-    
-    // check if it contains any not checked task list items
-    const hasOutstandingTasks = listItems.some(item => item.checked === false);
+    // check if this is an issue rather than pull event
+    if (context.event == 'issue_comment' && ! pr) {
+      // if so we need to make sure this is for a PR only
+      if (! context.payload.issue.pull_request) {
+        return;
+      }
+      // & lookup the PR it's for to continue
+      let response = await context.github.pulls.get(context.repo({
+        pull_number: context.payload.issue.number
+      }));
+      pr = response.data;
+    }
+
+    let hasOutstandingTasks = checkOutstandingTasks(pr.body);
+
+    // lookup comments on the PR
+    let comments = await context.github.issues.listComments(context.repo({
```

Comment:
What happens if these requests fail?

Reply:
Thanks @hjkatz great point! The requests only use Github's api but you're right they could certainly fail. 
Currently what will happen is it'll error, loudly in glitch for the logs. 
Thinking more about this, it could even error I guess if you don't have the new permissions granted. 
I think then the safest option is if these fail/error then we skip that check?

Reply:
IMO I think the opposite state would be desired, where the check is by default a failure/pending until the API works. This of course could be a problem when the API calls fail with bad permissions or with a bug in the code.

It's up to you though :)

Reply:
It's a tricky one, if someone's using this and they haven't granted the new permissions we need, I still want the app to work for them. Also if Github's having some issues, I'd rather not hold up releases, though I guess in that case it could be better to temp disable it as a required check when needed. 

I'll have a look through github's api docs and see if we can easily tell the difference between permission issues & actual api errors as then we can have best of both :)

Reply:
For what it's worth, I prefer the conservative approach @hjkatz suggests: If the app can't determine that things are successful (even if due to permissions issues) then it should report a failure. I'm not sure what other GitHub apps do in this situation, though.

Reply:
I agree with api errors, but I disagree about permissions though, a lot of people have already installed the app, this new PR asks for new permissions and if someone has already installed the app and does not want to accept the new permissions for whatever reason I would not want to force them to by breaking their app experience.

Reply:
Will this app do anything useful if they don't accept the permissions? My thinking is that, if the app is just going to always return a "pass" result in this case, then this isn't very useful and could be harmful in that it's misleading; in this case, they'd be better off uninstalling the app. But if the app can still do some useful work without these permissions, then I see your point.

Reply:
(But I'll also reiterate that I don't have much experience with GitHub apps in general, and particularly with maintaining backwards compatibility of them. I certainly see the rationale of not wanting to break things for existing users. So I don't want to argue my points too strongly, I just wanted to let you know my thinking on this.)

Reply:
I think you're thinking of new users, they wouldn't be able to install the app anyway without accepting all new permissions. 
But, existing users, they would want the app to continue working. 

The flow I believe is when we update our app permissions, existing users get an email to say our app is requesting new permissions and can click to accept them. However, they might not see this for a while, or may choose not to give the new permissions for whatever reason. 
Without the new permission, we can continue to work as we did before, what I'm thinking is the new api request that looks at the comments that requires the new permission can likely be wrapped in a try catch or detect otherwise that it failed (will investigate this workflow shortly) and will just continue if there's a permission error. 

This then allows new users to sign up with full new permissions, while allowing the existing app to continue working for existing installs the way it did before untill they accept the new permissions giving the app the new capabilities. 

Hope this clears this up.

Reply:
Okay, thanks for the reply. If it can be made to work the way you describe, then I agree that sounds like good behavior. Thanks!

Reply:
Back to it again now, found time between other projects now I'm back from vacation :)

Had a quick test and looks like actually the new issues permission I've added is only needed for the issue_comment event to be hooked up, checking `context.github.issues` appears to work with and without the new permission meaning the permission not being granted will only not recieve the new issue comment events, no errors to be caught the app side :+1: 

In terms of other possible errors, it's possible if github is having an issue the app could error, but the logs will catch those and from my testing the app doesn't crash, only errors loudly in the logs. 
For now I think that might actually be enough as what will happen is the check will fail to complete and not pass. 

Future improvement I think would be for it to have retry logic? But we can add that in a new PR I think as it'd be good to add this generically for all the api requests we make rather than just these new ones :)